File: java/03-find-max-in-list.java
// Пример программы нахождения максимального элемента в массиве из 20 чисел. 
public class Main {
    public static void main(String[] args) {
        final int LENGHT = 20;

        int m[] = new int[LENGHT];

        // Заполняем массив m из параметров запуска 
        // Нумерация входных параметров начинается с 0 индекса массива args
        for (int i = 0; i < LENGHT; i++) {
            // Получаем значения i-ого элемента массива
            m[i] = Integer.parseInt(args[i]);
        }

        // Находим максимальный элемент
        int max = m[0];
        for (int i = 1; i < LENGHT; i++) {
            if (m[i] > max) {
                max = m[i];
            }
        }

        // Вывод результата
        System.out.print(max);
    }
}

File: pascal/03-find-max-in-list.pas
{ Пример программы нахождения максимального элемента в массиве из 20 чисел.  }
uses sysutils;
const
    N = 20;
var 
    i   : Integer;
    m   : array[1..N] of Integer;
    max : Integer;
begin
    { Заполняем массив m из параметров запуска }
    { Нумерация входных параметров начинается с 1 индекса }
    for i := 1 to N do
    begin
        m[i] := StrToInt(ParamStr(i));
    end;
    
    { Находим максимальный элемент }
    max := m[1];
    for i := 2 to N do
    begin
        if (m[i] > max) then
            max := m[i];
    end;
    
    { Вывод результата }
    Writeln(max);
end.